pkg/rule/postgres.go

```diff
@@ -0,0 +1,498 @@
+package rule
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/lifesum/configsum/pkg/errors"
+	"github.com/lifesum/configsum/pkg/pg"
+)
+
+const (
+	pgRuleCreateSchema = `CREATE SCHEMA IF NOT EXISTS rule`
+	pgRuleCreateTable  = `
+		CREATE TABLE IF NOT EXISTS rule.rules(
+			id TEXT NOT NULL PRIMARY KEY,
+			active BOOLEAN NOT NULL,
+			activated_at TIMESTAMP WITHOUT TIME ZONE DEFAULT (now() AT TIME ZONE 'utc'),
+			buckets JSONB NOT NULL,
+			config_id TEXT NOT NULL,	
+			created_at TIMESTAMP WITHOUT TIME ZONE DEFAULT (now() AT TIME ZONE 'utc'),
+			criteria JSONB NOT NULL,
+			description TEXT NOT NULL,
+			end_time TIMESTAMP WITHOUT TIME ZONE DEFAULT (now() AT TIME ZONE 'utc'),
+			kind INT8 NOT NULL,
+			name TEXT NOT NULL,
+			start_time TIMESTAMP WITHOUT TIME ZONE DEFAULT (now() AT TIME ZONE 'utc'),
+			updated_at TIMESTAMP WITHOUT TIME ZONE DEFAULT (now() AT TIME ZONE 'utc')
+		)`
+	pgRuleDropTable = `DROP TABLE IF EXISTS rule.rules CASCADE`
+
+	pgRuleInsert = `
+		INSERT INTO
+		rule.rules(
+			id, 
+			active, 
+			activated_at, 
+			buckets, 
+			config_id, 
+			created_at, 
+			criteria, 
+			description,
+			end_time,
+			kind,
+			name,
+			start_time, 
+			updated_at) 
+			VALUES(
+				:id,
+				:active,
+				:activatedAt,
+				:buckets,
+				:configId,
+				:createdAt,
+				:criteria,
+				:description,
+				:endTime,
+				:kind,
+				:name,
+				:startTime,
+				:updatedAt
+		)`
+
+	pgRuleGet = `
```

Comment:
To make it more explicit we could encode what the criteria is that we get the rule by, e.g.: `pgRuleGetByName`.